src/code_23680.c

```diff
@@ -88,11 +108,64 @@ s32 is_point_within_region(s32 shape, f32 pointX, f32 pointY, f32 centerX, f32 c
 
 INCLUDE_ASM(s32, "code_23680", func_800490B4, s32 arg0, Enemy* arg1, f32 arg2, s32 arg3, s32 arg4);
 
-INCLUDE_ASM(s32, "code_23680", func_800493EC);
+s32 func_800493EC(Enemy* enemy, s32 arg1, f32 arg2, f32 arg3) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Npc* npc = get_npc_unsafe(enemy->npcID);
+    s32 ret;
+
+    if (arg1 >= 0) {
+        s32 temp;
+
+        if (arg1 != 0) {
+            temp = rand_int(arg1 + 1);
+        } else {
+            temp = 0;
+        }
+
+        if (temp == 0) {
+            f32 f1 = npc->pos.x;
+            f32 f2 = npc->pos.z;
+
+            add_vec2D_polar(&f1, &f2, arg3, 270.0f - npc->unk_34);
+
+            if (!(dist2D(f1, f2, playerStatus->position.x, playerStatus->position.z) <= arg2)) {
+                ret = 0;
+            } else {
+                ret = 1;
+            }
+        } else {
+            ret = 0;
+        }
+    } else {
+        ret = 0;
+    }
+
+    return ret;
```

Comment:
This function looks weird... have we tried removing `ret` and using `return` instead?